spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -129,7 +128,88 @@ def write_recording(recording, save_path, acquisition_name):
             description='acquisition_description'
         )
         nwbfile.add_acquisition(ephys_ts)
+
         if os.path.exists(save_path):
             os.remove(save_path)
         with NWBHDF5IO(save_path, 'w') as io:
             io.write(nwbfile)
+
+
+
+
+class NwbSortingExtractor(se.SortingExtractor):
+    def __init__(self, path):
+        try:
+            from pynwb import NWBHDF5IO
+            from pynwb import NWBFile
+            from pynwb.ecephys import ElectricalSeries
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("To use the Nwb extractors, install pynwb: \n\n"
+                                      "pip install pynwb\n\n")
+        #self._path = path
+        #with NWBHDF5IO(path, 'r') as io:
+        #    nwbfile = io.read()
+        #    NSX = se.NumpySortingExtractor()
+        se.RecordingExtractor.__init__()
+
+    @staticmethod
+    def write_sorting(sorting, save_path):
+        try:
+            from pynwb import NWBHDF5IO
+            from pynwb import NWBFile
+            from pynwb.ecephys import ElectricalSeries
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("To use the Nwb extractors, install pynwb: \n\n"
+                                      "pip install pynwb\n\n")
+        M = len(sorting.get_unit_ids())
+        fs = sorting.get_sampling_frequency()
+
+        #if NWB files already exists, just adds sorting data to it
+        if os.path.exists(save_path):
+            with NWBHDF5IO(save_path, 'r+') as io:
+                nwbfile = io.read()
+
+                #Stores spike times for each detected cell (unit)
+                for id in range(M):
+                    spkt = sorting.get_unit_spike_train(unit_id=id+1) / fs
+                    print('ID: ',id)
+                    print('Spkt: ', spkt)
+                    nwbfile.add_unit(
+                        id=id,
+                        spike_times=spkt,
+                        obs_intervals=None,
+                        electrodes=None,
+                        electrode_group=None,
+                        waveform_mean=None,
+                        waveform_sd=None
+                    )
+
+                io.write(nwbfile)
+
+        #if new NWB file does not exist, create it and add sorting data
+        else:
+            nwbfile = NWBFile(
+                session_description='',
+                identifier='',
+                session_start_time=datetime.now(),
+                experimenter='',
```

Comment:
Optional comments should not be ''. If no experimenter is provided you can just omit this argument. It's optional. That's better than passing an empty string. The same goes for several of these arguments.